arch/renesas/src/rx65n/rx65n_dtc.c

```diff
@@ -0,0 +1,1822 @@
+
+  dtc_dynamic_transfer_data_cfg_t *p_dtransfer_cfg =
+                           (dtc_dynamic_transfer_data_cfg_t *)dcfg;
+
+  /* Activation source */
+
+  dtc_activation_source_t act_source = (dtc_activation_source_t)src;
+
+  flags = enter_critical_section();
+
+  /* Store old value of DTCERn.DTCE bit. */
+
+  dtce_backup = ICU.DTCER[act_source].BIT.DTCE;
+
+  /* Disable the interrupt source. Clear the DTCER */
+
+  ICU.DTCER[act_source].BIT.DTCE = 0;
+
+  /* Store RRS and clear RRS */
+
+  rrs_backup =  DTC.DTCCR.BIT.RRS;
+  DTC.DTCCR.BIT.RRS = 0;
+
+  /* Get Transfer data pointer */
+
+  ptr = (uint32_t *)(dtchandle->vectortable + (4 * act_source));
+  p_transfer_data  = (dtc_transfer_data_t *) (*ptr);
+
+  ret = rx65n_dtc_validate_dynamic_params(p_dtransfer_cfg, p_transfer_data);
+  if (DTC_SUCCESS != ret)
+    {
+      return ret;
+    }
+
+  /* Update transfer data with address and counter */
+
+  for (i = 0; i < count; i++)
+    {
+      if (rx65n_dtc_set_dynamic_transfer_data(p_dtransfer_cfg,
+                                  p_transfer_data) != DTC_SUCCESS)
+        {
+          return DTC_ERR_INVALID_ARG;
+        }
+
+      p_dtransfer_cfg++;
+      p_transfer_data++;
+      count --;
+    }
+
+  /* Restore RRS bit */
+
+  DTC.DTCCR.BIT.RRS = rrs_backup;
+
+  /* Restore the DTCE bit. */
+
+  ICU.DTCER[act_source].BIT.DTCE = dtce_backup;
+
+  leave_critical_section(flags);
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: rx65n_dtc_setup_static_transferdata
+ *
+ * Description:
+ *   Configure and Creates the transfer data for a activation source.
+ *
+ * Input Parameters:
+ *    handle - DTC_HANDLE
+ *    src -Activation source
+ *    scfg -Pointer to contains the settings for Transfer data
+ *    pdt - pointer to data transfer
+ *    nchain - Number of chain transfer
+ *
+ * Returned Value:
+ *  dtc_err_t -DTC API error codes
+ *
+ ****************************************************************************/
+
+dtc_err_t rx65n_dtc_setup_static_transferdata(DTC_HANDLE handle, uint8_t src,
+                               uint32_t scfg, uint32_t pdt, uint32_t nchain)
+{
+  struct rx65n_dtc_s *dtchandle = (struct rx65n_dtc_s *)handle;
+
+  uint32_t *ptr = NULL;
+
+  int i;
+
+  /* Transfer Information count */
+
+  uint32_t count = nchain + 1;
+
+  /* Activation source */
+
+  dtc_activation_source_t act_source = (dtc_activation_source_t)src;
+  if ((act_source < DTC_MIN_VECT_NUMBER)
+              || (act_source > DTC_MAX_VECT_NUMBER))
+    {
+      return DTC_ERR_INVALID_ARG;
+    }
+
+  /* Static User Configuration */
+
+  dtc_static_transfer_data_cfg_t *p_stransfer_cfg =
+                          (dtc_static_transfer_data_cfg_t *)scfg;
+  if (p_stransfer_cfg == NULL)
+    {
+      return DTC_ERR_INVALID_ARG;
+    }
+
+  /* Allocate  memory in RAM for transfer information */
+
+  dtc_transfer_data_t *p_transfer_data = (dtc_transfer_data_t *)pdt;
```

Comment:
ditto

Reply:
Thank you for the comments.

The comments have been addressed in b0ae3fe.